app/workers/process_pdf_worker.rb

```diff
@@ -0,0 +1,23 @@
+class ProcessPdfWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: false
+
+  def perform(id)
+    presentation = Presentation.find(id)
+    images = Magick::Image.read(presentation.file.path) do
+      self.quality = 100
+      self.density = '300x300'
+    end
+
+    images.each_with_index do |image, i|
+      path = Rails.root.join("tmp/storage/#{presentation.file.identifier}-#{i}.png")
```

Comment:
file = Tempfile.new("file", "png")
file.write

# file.unlink for tempfiles after creation